src/ir/ty.rs

```diff
@@ -568,7 +568,8 @@ impl Type {
         let canonical_ty = ty.canonical_type();
         let kind = match ty.kind() {
             CXType_Unexposed if *ty != canonical_ty &&
-                                canonical_ty.kind() != CXType_Invalid => {
+                                canonical_ty.kind() != CXType_Invalid &&
+                                !canonical_ty.spelling().contains("type-parameter") => {
```

Comment:
Maybe some kind of comment or explanation here? This condition is not immediately clear to readers without all of the context, such as myself.

It is a state I often find myself in :)